app/src/main/java/com/jHerscu/clearskies/ui/MainActivity.kt

```diff
@@ -5,14 +5,23 @@ import android.os.Bundle
 import androidx.activity.ComponentActivity
 import androidx.activity.compose.setContent
 import androidx.activity.viewModels
+import androidx.compose.foundation.isSystemInDarkTheme
+import androidx.compose.runtime.getValue
 import androidx.core.splashscreen.SplashScreen.Companion.installSplashScreen
 import androidx.core.view.WindowCompat
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import com.jHerscu.clearskies.domain.repoInterface.LockedThemePref
+import com.jHerscu.clearskies.domain.useCase.preferences.ReadThemePreferencesUseCase
+import com.jHerscu.clearskies.domain.useCase.preferences.ThemePreferences
 import com.jHerscu.clearskies.ui.theme.AppTheme
 import dagger.hilt.android.AndroidEntryPoint
+import javax.inject.Inject
 
 @AndroidEntryPoint
 class MainActivity : ComponentActivity() {
 
+    @Inject protected lateinit var readThemePreferences: ReadThemePreferencesUseCase
```

Comment:
protected makes it effectively private here, however hilt doesn't allow injecting with private var